### PR TITLE
Fix RoundRobinPolicy

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Please outline the changes and how this PR fixes the issue.

### Why are the changes needed?

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/block/policy/RoundRobinPolicy.java [hunk 1]
```diff
@@ -35,10 +36,11 @@
 @NotThreadSafe
 public final class RoundRobinPolicy implements BlockLocationPolicy {
   private List<BlockWorkerInfo> mWorkerInfoList;
-  private int mIndex;
+  private static AtomicInteger mIndex = new AtomicInteger();
```
Comment (jiacheliu3) [inline core/client/fs/src/main/java/alluxio/client/block/policy/RoundRobinPolicy.java:39]: Does it make better sense if you just make this class a singleton?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Added  a high level comment, PTAL :)

File: core/client/fs/src/main/java/alluxio/client/block/policy/RoundRobinPolicy.java (hunk 1 above)
Comment (jiacheliu3) [reply core/client/fs/src/main/java/alluxio/client/block/policy/RoundRobinPolicy.java:39]: Basically using a static field to keep the state is a weird design, singleton is a better choice.

**Comment by maobaolong** [comment]
@jiacheliu3 I totally agree your suggestion, just close this PR

**Pull request closed without merging**